### DISCUSSION TITLE
what do you think of this other typescript zxcvbn implementation?

### DISCUSSION THREAD

**Discussion by paulverbeke** [zxcvbn-ts/zxcvbn#112] [Q&A]
Hello @MrWook ,

Hope you're doing well.
Could you tell me what do you think about this zxcvbn implementation https://github.com/trichards57/zxcvbn, also in typescript, and more or less still maintained ?

I ask you this because I needed to use zxcvbn in .NET for a project and I went and tried the most popular library, the most well maintained https://github.com/trichards57/zxcvbn-cs (which is a port of the typescript one), and after some testing I'm not entirely satisfied with it's results. 
For example Azerty12345* got a 3/4 score, that doesn't look right to me.

Does the original zxcvbn had bugs and problems in the scoring system ?
Should I make a .NET port of your typescript library though ? Would this add value or the original scoring system is still fine ?

**Comment by MrWook** [comment]
Hey @paulverbeke,

to start off, the typescript port you mentioned seems to have a problem because `Azerty12345*` has only a score of 2 in the original library. But the original library has some flaws too. I don't know the exact bugs but there were two or three. One is still open in the zxcvbn-ts repo as an issue.
For me it seems like this port is missing features like i18n and dictionaries per language which is a huge requested feature in the original library and is really needed for password security because the language that your user base is using is defining their used password.

In my opinion if you have project where you just wanna throw zxcvbn against you can use the .NET port that is already available. It's still better than those weird password rules: 8 chars 1 uppercase 1 lowercase and so on
If you have the time, ressources and the passion to do an updated .NET port with new features and bug fixes go for it.
One of the java ports https://github.com/GoSimpleLLC/nbvcxz has those features and bug fixes too. But the author did them a long time ago :D
